crates/runtime/src/dataaccelerator.rs

```diff
@@ -75,41 +74,166 @@ pub enum Error {
+
+        // Inject secrets from the user-supplied params.
+        // This will replace any instances of `${ store:key }` with the actual secret value.
+        for (k, v) in &acceleration_settings.params {
+            let secret = secret_guard.inject_secrets(k, ParamStr(v)).await;
+            params_with_secrets.insert(k.clone(), secret);
+        }
+
+        let params = Parameters::try_new(
+            &format!("accelerator {}", accelerator.name()),
+            params_with_secrets.into_iter().collect::<Vec<_>>(),
+            accelerator.prefix(),
+            secrets,
+            accelerator.parameters(),
+        )
+        .await
+        .context(AccelerationCreationFailedSnafu)?;
+
+        // Not all acceleration engines support creating tables with schemas so we include the schema as part of the table name.
+        // For example, Table {schema: "schema", table: "table_name"} is converted to Table {table: "schema.table_name"}.
+        let accelerated_table_name = TableReference::bare(table_name.to_string());
+
+        let mut external_table_builder = AcceleratorExternalTableBuilder::new(
+            accelerated_table_name,
+            Arc::clone(&schema),
+            engine,
+        )
+        .mode(acceleration_settings.mode)
+        .options(params)
+        .indexes(acceleration_settings.indexes.clone());
+
+        // If there are constraints from the federated table, then add them to the accelerated table
+        // and automatically configure upsert behavior for them. This can be overridden by the user.
+        if let Some(constraints) = constraints {
+            if !constraints.is_empty() {
+                external_table_builder = external_table_builder.constraints(constraints.clone());
+                let primary_keys: Vec<String> =
+                    get_primary_keys_from_constraints(constraints, &schema);
+                external_table_builder = external_table_builder
+                    .on_conflict(OnConflict::Upsert(ColumnReference::new(primary_keys)));
+            }
+        }
+
+        if let Some(on_conflict) =
+            acceleration_settings
+                .on_conflict()
+                .map_err(|e| Error::InvalidConfiguration {
+                    msg: format!("on_conflict invalid: {e}"),
+                })?
+        {
+            external_table_builder = external_table_builder.on_conflict(on_conflict);
+        };
+
+        match acceleration_settings.table_constraints(Arc::clone(&schema)) {
+            Ok(Some(constraints)) => {
+                if !constraints.is_empty() {
+                    external_table_builder = external_table_builder.constraints(constraints);
+                }
+            }
+            Ok(None) => {}
+            Err(e) => {
+                InvalidConfigurationSnafu {
+                    msg: format!("{e}"),
+                }
+                .fail()?;
+            }
+        }
```

Comment:
```suggestion
        if let Some(constraints) = acceleration_settings.table_constraints(Arc::clone(&schema))? {
            if !constraints.is_empty() {
                external_table_builder = external_table_builder.constraints(constraints);
            }
        }
```

the transparent error will also simplify this match